### ISSUE TITLE
mysterious clang-tidy bugprone-use-after-move warning in obj-c block

### ISSUE THREAD

**Issue by zmodem** [llvm/llvm-project#59130]
Consider:

```console
$ cat /tmp/a.mm
#include <utility>

struct S {
  void foo() const;
  bool b;
};

void run_block(void (^block)());

void bar(S s) {
  run_block(^{
    if (s.b) {
      std::move(s).foo();
    }
  });
}

$ build/bin/clang-tidy /tmp/a.mm -checks=bugprone-use-after-move -- -fblocks
1 warning generated.
/tmp/a.mm:12:9: warning: 's' used after it was moved [bugprone-use-after-move]
    if (s.b) {
        ^
/tmp/a.mm:13:20: note: move occurred here
      std::move(s).foo();
                   ^
/tmp/a.mm:12:9: note: the use and move are unsequenced, i.e. there is no guarantee about the order in which they are evaluated
    if (s.b) {
        ^
```

Is this a false positive, or am I missing something?

(Based on http://crbug.com/1385107)

**Comment by zmodem** [comment]
This still reproduces at today's head revision.

@5chmidti @PiotrZSL do you know someone who could take a look at this?
